Sources/Interaction/Manipulators/MouseCameraUnicamRotateManipulator/index.js

```diff
@@ -0,0 +1,368 @@
+
+      const atV = camera.getViewPlaneNormal();
+      const upV = camera.getViewUp();
+      const rightV = [];
+      vtkMath.cross(upV, atV, rightV);
+      vtkMath.normalize(rightV, interactor);
+
+      //
+      // The following two tests try to prevent chaotic camera movement
+      // that results from rotating over the poles defined by the
+      // "WorldUpVector".  The problem is the constraint to keep the
+      // camera's up vector in line w/ the WorldUpVector is at odds with
+      // the action of rotating over the top of the virtual sphere used
+      // for rotation.  The solution here is to prevent the user from
+      // rotating the last bit required to "go over the top"-- as a
+      // consequence, you can never look directly down on the poles.
+      //
+      // The "0.99" value is somewhat arbitrary, but seems to produce
+      // reasonable results.  (Theoretically, some sort of clamping
+      // function could probably be used rather than a hard cutoff, but
+      // time constraints prevent figuring that out right now.)
+      //
+      if (model.useWorldUpVec) {
+        const OVER_THE_TOP_THRESHOLD = 0.99;
+        if (vtkMath.dot(upVec, atV) > OVER_THE_TOP_THRESHOLD && rDist < 0) {
+          rDist = 0;
+        }
+        if (vtkMath.dot(upVec, atV) < -OVER_THE_TOP_THRESHOLD && rDist > 0) {
+          rDist = 0;
+        }
+      }
+
+      rotateCamera(camera, ...center, ...rightV, rDist);
+
+      if (
+        model.useWorldUpVec &&
+        !vtkMath.areEquals(upVec, camera.getViewPlaneNormal())
+      ) {
+        camera.setViewUp(...upVec);
+      }
+
+      model.previousPosition = position;
+
+      renderer.resetCameraClippingRange();
+      updateAndRender(interactor);
+    }
+  };
+
+  //----------------------------------------------------------------------------
+  // Public API methods
+  //----------------------------------------------------------------------------
+  publicAPI.onButtonDown = (interactor, renderer, position) => {
+    model.buttonPressed = true;
+    model.startPosition = position;
+    model.previousPosition = position;
+
+    // Finds the point under the cursor.
+    // Note: If no object has been rendered to the pixel (X, Y), then
+    // vtkPicker will return a z-value with depth equal
+    // to the distance from the camera's position to the focal point.
+    // This seems like an arbitrary, but perhaps reasonable, default value.
+    let selections = null;
+    if (model.useHardwareSelector) {
+      model.selector.attach(interactor.getView(), renderer);
+
+      model.selector.setArea(position.x, position.y, position.x, position.y);
+      selections = model.selector.select();
+    }
+
+    if (selections && selections.length !== 0) {
+      model.downPoint = selections[0].getProperties().worldPosition;
+    } else {
+      model.picker.pick([position.x, position.y, position.z], renderer);
+      model.downPoint = model.picker.getPickPosition();
```

Comment:
This method seems to pick points that are way out of the model. Can you try using picker.getPickedPositions() instead ? Usually sorting by distance to camera lets you get the best point.